kernel/core/strings/stringfunctions.hpp

```diff
@@ -2760,6 +2760,40 @@ static StringVector strfnc_sha256(StringFuncArgs& funcArgs)
 }
 
 
+/////////////////////////////////////////////////
+/// \brief Implementation of beginswith() function
+///
+/// \param funcArgs StringFuncArgs&
+/// \return StringVector
+///
+/////////////////////////////////////////////////
+static StringVector strfnc_beginswith(StringFuncArgs& funcArgs)
+{
+    return funcArgs.sArg2.view().length() <= funcArgs.sArg1.view().length()
+        && equal(funcArgs.sArg2.view().begin(),
+                 funcArgs.sArg2.view().end(),
+                 funcArgs.sArg1.view().begin()
+                 );

```

Comment:
Interesting method. A bit hard to read, I'm a bit more familiar with `funcArgs.sArg2.view() == funcArgs.sArg1.view().subview(0, funcArgs.sArg2.view().length())`, but that seems to work as well.